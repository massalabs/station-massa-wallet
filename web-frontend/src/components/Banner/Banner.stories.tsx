import Banner, { BannerProps } from './Banner';

export default {
  component: Banner,
  title: 'Banner',
};

const Template = (args: BannerProps) => <Banner {...args} />;

export const DefaultBanner = () => (
  <Template>Welcome to My Awesome React TypeScript App!</Template>
);
