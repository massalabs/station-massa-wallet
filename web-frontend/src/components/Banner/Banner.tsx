import React from 'react';

export interface BannerProps {
  children: React.ReactNode;
}

// TODO: urban font family

export default function Banner(props: BannerProps) {
  return (
    <h1 className="text-center text-4xl font-semibold text-black leading-10 tracking-massa">
      {props.children}
    </h1>
  );
}
