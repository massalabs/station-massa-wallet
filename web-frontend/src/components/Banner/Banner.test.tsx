import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import Banner from './Banner';

afterEach(cleanup);

describe('Banner', () => {
  it('should render a banner with the given text', () => {
    const message = 'Welcome to My Awesome React TypeScript App!';
    const banner = <Banner>{message}</Banner>;

    // snapshot test
    const componentSnapchat = renderer.create(banner);
    const tree = componentSnapchat.toJSON();
    expect(tree).toMatchSnapshot();

    // DOM test
    const componentDom = render(banner);
    expect(componentDom.getByText(message)).toBeTruthy();
  });
});
