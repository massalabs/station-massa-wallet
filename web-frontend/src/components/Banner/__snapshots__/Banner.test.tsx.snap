// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should render a banner with the given text 1`] = `
<h1
  className="text-center text-4xl font-semibold text-black leading-10 tracking-massa"
>
  Welcome to My Awesome React TypeScript App!
</h1>
`;
